src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -323,7 +326,48 @@ public String getLaunchURLForCaseId(final UUID caseId, LaunchRequestDTO requestP
         .with("formType", formType)
         .info("Have generated new questionnaireId");
 
-    // Finally, build the url needed to launch the survey
+    if (caseType == CaseType.CE && !individual && "CE".contentEquals(formType)) {
+      rejectInvalidLaunchCombinations(caseDetails.getRegion(), caseDetails.getAddressLevel());
+    }
+
+    String eqUrl = createLaunchUrl(formType, caseDetails, requestParamsDTO, questionnaireId);
+    publishSurveyLaunchedEvent(caseDetails.getId(), questionnaireId, requestParamsDTO.getAgentId());
+    return eqUrl;
+  }
+
+  private void rejectInvalidLaunchCombinations(String region, String addressLevel)
```

Comment:
Is there any way for us to deal with the enum type here rather than the string representation of it's name? It seems a waste of an enum to turn it into a string and do comparisons on that.

Reply:
I  would normally do that, but sadly there is not exact match. I've talked to Ameet about it and the string can bring in Scotland, and blank is another tolerated thing. So if I did the enum conversion first I would have to test for something like null in the reject method which I'm not sure is cleaner.